### PR TITLE
cmake format

### PR DESCRIPTION
**Pull request by thewtex**
- ENH: Add cmake-format config and script
- STYLE: Apply ./Utilities/Maintenance/cmake-format.bash --tracked

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Beautiful @thewtex :100:. A few questions/comments:

- Will this format any change made to the CMake files locally prior to a commit? If not, do you think it can be done within the scope of this PR?
- Will this trigger a GHA linter workflow for CMake files? If not, do you think it can be done within the scope of this PR?
- IIRC, in the early days we adopted/CMake used to use uppercases for its commands, but then switched to lowercases. Will this change them to lowercases as well ? If not, can that be done through the config file.
- Can we add a short description to the body of both commit messages?

File: CMake/CppcheckTargets.cmake [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
-# Original Author:
-# 2009-2010 Ryan Pavlik <rpavlik@iastate.edu> <abiryan@ryand.net>
-# http://academic.cleardefinition.com
+# Original Author: 2009-2010 Ryan Pavlik <rpavlik@iastate.edu> <abiryan@ryand.net> http://academic.cleardefinition.com
 # Iowa State University HCI Graduate Program/VRAC
 #
-# Addition:
-# 2011 Arnaud Gelas <arnaud_gelas@hms.harvard.edu>
-# Harvard Medical School
+# Addition: 2011 Arnaud Gelas <arnaud_gelas@hms.harvard.edu> Harvard Medical School
```
Comment (dzenanz) [inline CMake/CppcheckTargets.cmake:15]: Do we want this soft of reformatting of the comments? Can we disable it?

File: CMake/CppcheckTargets.cmake [hunk 2]
```diff
@@ -280,35 +255,24 @@ function(add_cppcheck _name)
...
-          ${CPPCHECK_EXECUTABLE}
-          ${CPPCHECK_QUIET_ARG}
-          ${CPPCHECK_TEMPLATE_ARG}
-          ${_cppcheck_args}
-          ${_files}
-        WORKING_DIRECTORY
-          "${CMAKE_CURRENT_SOURCE_DIR}"
-        COMMENT
-          "${_name}_cppcheck: Running cppcheck on target ${_name}..."
-       V ERBATIM)
```
Comment (dzenanz) [inline CMake/CppcheckTargets.cmake:311]: `V ERBATIM` with space seems like a bug. Does it affect anything?

File: CMake/ExternalData.cmake [hunk 3]
```diff
@@ -415,19 +405,18 @@ function(ExternalData_add_target target)
...
           COMMENT "Generating ${file}"
           OUTPUT "${file}"
-          COMMAND ${CMAKE_COMMAND} -Drelative_top=${CMAKE_BINARY_DIR}
-                                   -Dfile=${file} -Dname=${name}
-                                   -DExternalData_ACTION=local
-                                   -DExternalData_CONFIG=${config}
-                                   -P ${_ExternalData_SELF}
-          MAIN_DEPENDENCY "${name}"
-          )
+          COMMAND ${CMAKE_COMMAND} -Drelative_top=${CMAKE_BINARY_DIR} -Dfile=${file} -Dname=${name}
```
Comment (dzenanz) [inline CMake/ExternalData.cmake:417]: This was more readable previously. Can this be configured?

File: Modules/IO/TransformMINC/CMakeLists.txt [hunk 4]
```diff
@@ -2,11 +2,8 @@ project(ITKIOTransformMINC)
 set(ITKIOTransformMINC_LIBRARIES ITKIOTransformMINC)
 
 if(NOT ITK_USE_SYSTEM_HDF5)
-  include_directories(
-    "${ITKHDF5_SOURCE_DIR}/src/itkhdf5"
-    "${ITKHDF5_SOURCE_DIR}/src/itkhdf5/src"
-    "${ITKHDF5_BINARY_DIR}/src/itkhdf5"
-    )
+  include_directories("${ITKHDF5_SOURCE_DIR}/src/itkhdf5" "${ITKHDF5_SOURCE_DIR}/src/itkhdf5/src"
```
Comment (dzenanz) [inline Modules/IO/TransformMINC/CMakeLists.txt:5]: This was more readable previously. Could we enforce one dir per line?

**Review by dzenanz — COMMENTED** [review]
Generally seems helpful, but I would like the style options adjusted somewhat if possible.

File: CMake/CppcheckTargets.cmake (hunk 1 above)
Comment (thewtex) [reply CMake/CppcheckTargets.cmake:15]: We can try disabling comment re-formatting.

File: CMake/CppcheckTargets.cmake (hunk 2 above)
Comment (thewtex) [reply CMake/CppcheckTargets.cmake:311]: This is an unrelated issue: patch here: https://github.com/InsightSoftwareConsortium/ITK/pull/4234

File: Modules/IO/TransformMINC/CMakeLists.txt (hunk 4 above)
Comment (thewtex) [reply Modules/IO/TransformMINC/CMakeLists.txt:5]: I think we can make one argument per line more relevant per configuration.

File: CMake/ExternalData.cmake (hunk 3 above)
Comment (thewtex) [reply CMake/ExternalData.cmake:417]: I tweaked the configuration. Also, I excluded this file, which is tracked externally.

**Review by dzenanz — APPROVED** [review]
While some changes are still a mixed bag, this is overall a noticeable improvement.

**Comment by jhlegarreta** [comment]
>> Will this format any change made to the CMake files locally prior to a commit? If not, do you think it can be done within the scope of this PR?
>> Will this trigger a GHA linter workflow for CMake files? If not, do you think it can be done within the scope of this PR?

> I will look at adding it to pre-commit hooks and GitHub Actions in follow-up work.

> It has support for pre-commit. I think we should consider adopting it, but that is a bigger effort.

Thanks Matt. I'll open the corresponding issue to keep track of it.

>> IIRC, in the early days we adopted/CMake used to use uppercases for its commands, but then switched to lowercases. Will this change them to lowercases as well ? If not, can that be done through the config file.

> Yes, it takes care of the command case, in addition to many other nagging items! 💅

Beautiful :100:.

>> Can we add a short description to the body of both commit messages?

> Yes, done. I also added our *.wrap for formatting.

Thanks Matt. You rock.

**Review by jhlegarreta — APPROVED** [review]

**Pull request merged**